src/ape/api/providers.py

```diff
@@ -1316,6 +1345,76 @@ def _find_txn_by_account_and_nonce(
                     stop_block,
                 )
 
+    def poll_blocks(self, stop_block: Optional[int] = None, required_confirmations: Optional[int] = None, new_block_timeout: Optional[int] = None) -> Iterator[BlockAPI]:
```

Comment:
i believe we are going to need to update the `ChainManager` methods (or the `QueryManager` methods, or both) to delegate to these new methods

Reply:
Updated ChainManager's poll methods to use the provider poll methods for non-historical data